src/material/icon/icon.ts

```diff
@@ -208,9 +223,25 @@ export class MatIcon extends _MatIconBase implements OnInit, AfterViewChecked, C
       elementRef: ElementRef<HTMLElement>, private _iconRegistry: MatIconRegistry,
       @Attribute('aria-hidden') ariaHidden: string,
       @Inject(MAT_ICON_LOCATION) private _location: MatIconLocation,
-      private readonly _errorHandler: ErrorHandler) {
+      private readonly _errorHandler: ErrorHandler,
+      @Optional() @Inject(MAT_ICON_DEFAULT_OPTIONS)
+      defaults?: MatIconDefaultOptions) {
     super(elementRef);
 
+    if (defaults) {
+      if (defaults.defaultColor) {
+        this.defaultColor = defaults.defaultColor;
+      }
+
+      if (defaults.color || defaults.defaultColor) {
+        this.color = defaults.color || defaults.defaultColor;
+      }
+
+      if (defaults.fontSet) {
+        this.fontSet = defaults.fontSet;
+      }
+    }
```

Comment:
I think this initialization could be folded into the existing property declarations, e.g.
```ts
private _fontSet: string = this._defaults?.fontSet;
```

Reply:
But that requires type changing from `string` to `string | undefined` for `_fontSet` property and `fontSet` getter/setter.

Reply:
Would it be behaviorally equivalent to fall back to the empty string?
```ts
private _fontSet: string = this._defaults?.fontSet ?? '';
```

Reply:
done

Reply:
@jelbourn, looks like some tests are failing now because of this. Do you prefer me to change test scenarios or revert to `undefined` instead of an empty string?

Reply:
Is that's what the tests had before then `undefined` is probably the least churn